impress-org/givewp-next-gen
# Feature: add donor comments block

## Description



This adds the Donor Comments block to our core blocks.

## Affects



- This new block adds a textarea field that integrates with the core donate controller to update the comment property on the donation model
- A form migration step was included
- The block will also be added upon new form if enabled in global/default options

## Visuals


<img width="1059" alt="Screenshot 2023-08-09 at 3 28 29 PM" src="https://github.com/impress-org/givewp-next-gen/assets/10138447/f4d4eb8b-3bc1-49cc-be26-fbf25dbd190b">


<img width="712" alt="Screenshot 2023-08-09 at 3 28 36 PM" src="https://github.com/impress-org/givewp-next-gen/assets/10138447/8195fc1c-78dd-4220-8aa5-e760b144fa65">


## Testing Instructions



- Add a donor comments block to a v3 form.
- Interact with the settings
- Go to the front-end and fill in the field
- Complete the donation and check the back-end to make sure it persists.
- You should also see a line on the confirmation page.

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205223433088674
